go/consensus/tendermint/apps/supplementarysanity/checks.go

```diff
@@ -149,10 +149,9 @@ func checkStaking(state *iavl.MutableTree, now epochtime.EpochTime) error {
 	if err != nil {
 		return fmt.Errorf("Delegations: %w", err)
 	}
-	for acct, delegations := range delegationses {
-		err = staking.SanityCheckDelegations(st.Account(acct), delegations)
-		if err != nil {
-			return fmt.Errorf("SanityCheckDelegations %s: %w", acct, err)
+	for id, delegations := range delegationses {
+		if err = staking.SanityCheckDelegations(id, st.Account(id), delegations); err != nil {
+			return err
```

Comment:
:information_source: responsibility for reporting which account in the error is pushed down to SanityCheckDelegations. This PR makes similar changes for other functions too.